rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -78,35 +79,20 @@ export PAI_CONTAINER_HOST_PORT="$(cut -f 1 -d"," <<< $PAI_CONTAINER_HOST_http_PO
 export PAI_CONTAINER_SSH_PORT="$(cut -f 1 -d"," <<< $PAI_CONTAINER_HOST_ssh_PORT_LIST)"
 
 task_role_no={{{ idx }}}
-# Touch a container id file in "APP_ID-TASK_ROLE_NO-TASK_INDEX-CONTAINER_HOST_IP-CONTAINER_HOST_PORT" format on hdfs
-# To communicate with other containers, add APP_ID as prefix to differentiate attempts with same job name
-hdfs dfs -touchz $HDFS_LAUNCHER_PREFIX/$PAI_USER_NAME/$PAI_JOB_NAME/tmp/$APP_ID-$task_role_no-$PAI_TASK_INDEX-$PAI_CONTAINER_HOST_IP-$PAI_CONTAINER_HOST_PORT || exit 1
-while [ `hdfs dfs -ls $HDFS_LAUNCHER_PREFIX/$PAI_USER_NAME/$PAI_JOB_NAME/tmp/ | grep "/$PAI_JOB_NAME/tmp/$APP_ID-" | wc -l` -lt  $PAI_JOB_TASK_COUNT ]; do
-  printf "%s %s\n" "[INFO]" "Waiting for other containers ..."
-  sleep 10
-done
-hdfs dfs -ls $HDFS_LAUNCHER_PREFIX/$PAI_USER_NAME/$PAI_JOB_NAME/tmp/ \
-  | grep "/$PAI_JOB_NAME/tmp/$APP_ID-" \
-  | grep -oE "[^/]+$" \
-  | sed -e "s/^$APP_ID-//g" \
-  | sort -n -k 2 -t"-" \
-  > ContainerList
-if [ "$(cat ContainerList | wc -l)" -ne $PAI_JOB_TASK_COUNT ]; then
-  printf "%s %s\n%s\n%s\n\n" \
-    "[ERROR]" "ContainerList" \
-    "$(cat ContainerList)" \
-    "$(cat ContainerList | wc -l) containers are available, not equal to $PAI_JOB_TASK_COUNT, exit ..."
-  exit 2
-fi
 
-export PAI_CURRENT_TASK_ROLE_CURRENT_TASK_INDEX=$((`cat ContainerList | grep "^$task_role_no-" | grep -n "$task_role_no-$PAI_TASK_INDEX-$PAI_CONTAINER_HOST_IP-$PAI_CONTAINER_HOST_PORT" | cut -d ":" -f 1`-1))
-task_role_list=(${PAI_JOB_TASK_ROLE_LIST//,/ })
-for i in `seq 0 $((PAI_JOB_TASK_ROLE_COUNT-1))`; do
-  host_list=`cat ContainerList | grep "^$i-" | cut -d "-" -f 3-4 | tr "-" ":" | sed -e :E -e "N;s/\n/,/;tE"`
-  export PAI_TASK_ROLE_${i}_HOST_LIST=$host_list
-  export PAI_TASK_ROLE_${task_role_list[$i]}_HOST_LIST=$host_list
+
+# Get container host list from framework launcher webservice
+framework_status=$(curl -X GET -H "Accept: application/json" {{{ statusUri }}} | jq -r ".aggregatedFrameworkStatus")
```

Comment:
>framework_status [](start = 0, length = 16)

you still need to call the launcher server in a while loop until the launcher starts all the containers.

Reply:
ok, this assumes gang scheduling is used,